/*
 * Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package nvidia.claraviz.volumestream;

import "nvidia/claraviz/core/camera.proto";

option java_package = "com.nvidia.claraviz.volumestream";

// Service that implements the Volume Stream Render Server.
service RenderServer
{
    // Sets the 3D viewing transformation
    rpc Camera(nvidia.claraviz.core.CameraRequest) returns (nvidia.claraviz.core.CameraResponse) {}

    // dummy
    rpc Dummy(DummyRequest) returns (DummyResponse) {}
}

// Message sent by the client for the `Dummy` method.
message DummyRequest {}

// Message returned to the client by the `Dummy` method.
message DummyResponse {}
